eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentPacket.scala

```diff
@@ -147,7 +148,12 @@ object IncomingPaymentPacket {
                 // blinding point and use it to derive the decryption key for the blinded trampoline onion.
                 decryptOnion(add.paymentHash, privateKey, trampolinePacket).flatMap {
                   case DecodedOnionPacket(innerPayload, Some(next)) => validateNodeRelay(add, payload, innerPayload, next)
-                  case DecodedOnionPacket(innerPayload, None) => validateTrampolineFinalPayload(add, payload, innerPayload)
+                  case DecodedOnionPacket(innerPayload, None) =>
+                    if (innerPayload.get[OutgoingBlindedPaths].isDefined) {
+                      Left(InvalidOnionPayload(UInt64(66102), 0)) // Trampoline to blinded paths is not yet supported.
+                    } else {
+                      validateTrampolineFinalPayload(add, payload, innerPayload)
+                    }
```

Comment:
nit: we'll probably want to extract the `OutgoingBlindedPaths` when actually using them, otherwise you'll probably need to do a `.get` later or duplicate the `.isDefined`?

```suggestion
                  case DecodedOnionPacket(innerPayload, None) => innerPayload.get[OutgoingBlindedPaths] match {
                    case Some(_) => Left(InvalidOnionPayload(UInt64(66102), 0)) // Trampoline to blinded paths is not yet supported.
                    case None => validateTrampolineFinalPayload(add, payload, innerPayload)
                  }
```

Reply:
This `if` is only to decide which `validate` function to call, there is no need to extract the paths. The `.isDefined` is currently duplicated in the `validate` function, I guess we could remove the second check but then we'll a function named `validate` that assumes some pre-validation.